tests/ocp/sriov/README.md

```diff
@@ -0,0 +1,550 @@
+export ECO_TEST_LABELS="sriov && !skip-in-ci"
+
+# Run specific test ID (test IDs can be used as labels)
+export ECO_TEST_LABELS="12345"
+
+# Run multiple test IDs
+export ECO_TEST_LABELS="12345 || 12346"
+```
+
+## Common Patterns
+
+### Resource Creation Pattern
+
+```go
+resource := builder.NewBuilder(APIClient, name, namespace)
+resource.WithLabel("key", "value")
+resourceObject, err := resource.Create()
+Expect(err).ToNot(HaveOccurred(), "Failed to create resource")
+```
+
+### Resource Cleanup Pattern
+
+```go
+DeferCleanup(func() {
+    err := resource.Delete()
+    Expect(err).ToNot(HaveOccurred(), "Failed to delete resource")
+})
+```
+
+### Validation Pattern
+
+```go
+Eventually(func() bool {
+    obj, err := builder.Pull(APIClient, name, namespace)
+    if err != nil {
+        return false
+    }
+    return obj.IsReady()
+}, timeout, interval).Should(BeTrue(), "Resource should be ready")
+```
+
+### Test Data Pattern
+
+```go
+type testData struct {
+    policy  *sriov.PolicyBuilder
+    network *sriov.NetworkBuilder
+    pod     *pod.Builder
+}
+
+var testResources []testData
+
+BeforeEach(func() {
+    testResources = []testData{}
+})
+```
+
+## Pre-Submit Checklist
+
+Before submitting a test case for review, ensure:
+
+- [ ] Test follows directory structure conventions
+- [ ] All required labels are defined in `tsparams/consts.go`
+- [ ] Test IDs are included using `reportxml.ID()`
+- [ ] Resources are properly cleaned up in `AfterEach` or `AfterAll`
+- [ ] Error handling is comprehensive
+- [ ] Timeouts use constants from `tsparams`
+- [ ] `By()` statements document test steps
+- [ ] Reporter is configured for failure reporting
+- [ ] Environment variables are documented in README
+- [ ] Test can run independently (unless using `Ordered`)
+- [ ] Code passes linting (`make lint`)
+- [ ] Test descriptions are clear and descriptive
+- [ ] README is updated with new features or environment variables
+- [ ] **All API calls use eco-goinfra packages** (no raw Kubernetes client calls)
+- [ ] **Reusable functions are placed in upper-level `tests/internal/` folder**
+- [ ] **Commit message includes team prefix** (e.g., `ocp-sriov: description`)
+
+## Additional Resources
+
+### Framework Documentation
+
+- [Project README](../../../README.md): General project information
+- [CNF Network README](../../cnf/core/network/README.md): Network test patterns
+- [Ginkgo Documentation](https://onsi.github.io/ginkgo): Ginkgo framework reference
+
+### Eco-goinfra Packages
+
+- [Eco-goinfra README](https://github.com/rh-ecosystem-edge/eco-goinfra#readme): Package documentation
+- [SR-IOV Package](https://github.com/rh-ecosystem-edge/eco-goinfra/tree/main/pkg/sriov): SR-IOV resource management
+
+### Best Practices References
+
+- Review existing test implementations in `tests/cnf/core/network/sriov/tests/`
+- Follow patterns from other test suites in the project
+- Consult team documentation for domain-specific guidelines
+
```

Comment:
I'm missing an example of how a test that's hard to fit in, fits in.

Go to each team involved, ask them about a test, and add one of those tests using _the same_ framework. 

If, while you're doing this, you hit roadblocks that cause you think "but this doesn't fit", then you're on the right track. It's exactly those that we need to reshape in a way that they do fit.